Automattic/wp-calypso
# Fix: site thumbnails update state when unmount

Fixes https://github.com/Automattic/wp-calypso/issues/73163

## Proposed Changes

This pr proposes early return when the component is unmounted and thus avoiding updating the state, creating the following warning:
```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
```

## Testing Instructions



1. Copy a text (eg: `abcdefghijklmno` ). (You need to make sure that the text doesn't let any site visible under your sites list after a search )
2. Load Sites page ( by visiting `/sites` )
3. Quickly paste the text you copied.
4. Ensure that you don't get the following warning

```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
```

## Pre-merge Checklist
